src/Raven.Pal/src/posix/getsysteminformation.c

```diff
@@ -48,3 +63,34 @@ rvn_get_path_disk_space(const char* path, uint64_t* total_free_bytes, uint64_t*
 
     return SUCCESS;
 }
+
+EXPORT int32_t
+rvn_get_path_disk_stats(const char * path, struct IO_STATS* io_stats, int32_t* detailed_error_code)
+{
+    struct stat stats;
+	*detailed_error_code = 0;
+    int minorValue, majorValue;
+    char dfile[MAX_PF_NAME];
+
+    /* stat() returns 0 on successful operation,
+    otherwise returns -1 if unable to get file properties.*/
+    if (stat(path, &stats) != 0)
+        goto error;
+
+    majorValue = major(stats.st_dev);
+    minorValue = minor(stats.st_dev);
+
+    /* Read stats for device */
+    snprintf(dfile, sizeof(dfile), "%s/%s/%d:%d/%s", SLASH_SYS, __DEV_BLOCK, majorValue, minorValue, S_STAT);
+    dfile[sizeof(dfile) - 1] = '\0';
```

Comment:
what file are you actually reading here (add comment)

Reply:
/sys/block/<dev>/stat - https://www.kernel.org/doc/Documentation/block/stat.txt

I mimicked the implementation from `iostat` tool
https://github.com/sysstat/sysstat/blob/master/iostat.c#L415

Reply:
Yes, add that as a comment, in 2 years, we'll not know